python/pyspark/sql/types.py

```diff
@@ -1020,10 +1020,19 @@ def _infer_type(obj):
         return dataType()
 
     if isinstance(obj, dict):
-        for key, value in obj.items():
-            if key is not None and value is not None:
-                return MapType(_infer_type(key), _infer_type(value), True)
-        return MapType(NullType(), NullType(), True)
+        from pyspark.sql.session import SparkSession
+        if (SparkSession._activeSession.conf.get(
+                "spark.sql.pyspark.inferNestedStructByMap").lower() == "true"):
```

Comment:
Can you pass a bool argument to `_infer_type` (and `_infer_schema`), and access to the configuration value like https://github.com/apache/spark/blob/master/python/pyspark/sql/pandas/conversion.py#L79?